src/Network/OAuth/OAuth2/Compat.hs

```diff
@@ -48,14 +68,14 @@ fetchAccessToken
     :: Manager
     -> OAuth2
     -> ExchangeToken
-    -> IO (OAuth2Result Errors OAuth2Token)
+    -> IO (OAuth2Result OAuth2Token)
```

Comment:
Don't changes in our own signatures like this mean we have to be a major version bump (`0.8.0.0`)?

Reply:
Yep, you're right. Will change.

Reply:
I guess also, is there any way to avoid it?

Reply:
It's tricky because the <2.7.0 error type is gone... As linked elsewhere, we re-throw that error.

So from a compat perspective I'm not really sure how to get around that (albeit implicit in an error's type) interface change.

Reply:
Do you truly mean "throw", or return as a potential `Left`?

Reply:
If we introduce our own `Error` that works the same to any possible use case (i.e. a pattern match), I'd consider that backwards compatible. It doesn't matter if we supply it now that the library does not.

It's a separately question how much we want to bend over backwards to continue to avoid v0.8 (or even v1.0). If we think we're at that point to bump we should consider moving our lower bound on hoauth2 up and ditching _all_ the CPP. Yeah?

Reply:
So we do a true throw but it looks like we catch it before it gets to possible downstream consumers.

It looks like this handling results in a 
 - log message (this will change but it's just a log, right? Right? A log format shouldn't change our interface, right?)
 - HTML page containing `dispatchErrorMessage` which doesn't care about the underlying, expecific error

So, I'll take a stab at making this backwards compatible.

Reply:
Looks like `OAuth2Error Errors` leaks to higher stuff than just the handler I'm looking at :( (e.g. stuff in our Prelude)... Not sure how promising this is feeling... I'lll see if there's something obvious but I'm leaning more towards the "ditch the CPP" approach.

Reply:
I'm curious why it bubbling up higher making it feel any less promising. Isn't the question just if we can preserve all the same interfaces or not? How many things rely on those interfaces doesn't really change the work to be done right here, does it?

Reply:
Talked it through offline. These concerns should all be internal to the package and should not affect downstream consumers.